### PR TITLE
Rubocop

### PR DESCRIPTION
**Pull request by munckymagik**
Right, I admit I've done it. I'm human; I want Rubocop around to run as a guide when I mess up and use the wrong code style, but I also want a clean slate to work from so we can really see when we introduce new issues.

Most things were fixed automatically. Some things I fixed manually. It wasn't a big deal.

I'm proposing for **this** project we disable `Metrics/AbcSize` and `Style/NumericLiterals`, but of course I'm open to opinions on this. Rationale for disabling these two is recorded in a comments in `.rubocop.yml`.

@neoeno @JiggyPete

### CODE REVIEW

**Comment by munckymagik** [comment]
... and @neoeno don't give me any of that "Nunca haras ese pull request, y lo sabes" shizzle

**Comment by neoeno** [comment]
image

File: Gemfile [hunk 1]
```diff
@@ -31,6 +31,7 @@ gem 'uglifier', '>= 1.3.0'
 gem 'unicorn'
 
 group :test, :development do
+  gem 'rubocop'
```
Comment (neoeno) [inline Gemfile]: supposed to be alpha order I think

File: README.md [hunk 2]
```diff
@@ -72,3 +72,11 @@ To run the javascript tests:
 ```
 $ node_modules/.bin/karma start spec/javascripts/karma.conf.js --single-run=true
 ```
+
+# Style Checking
+
+To run the Rubocop style checker:
+
+```sh
+$ bundle exec rubocop -DSc .rubocop.yml
```
Comment (neoeno) [inline README.md]: I _think_ `.rubocop.yml` is the default. Can't find a source though.

File: app/forms/admin/move_advisers_form.rb [hunk 3]
```diff
@@ -33,9 +33,9 @@ def subsidiaries
     private
 
     def destination_firm_fca_number_exists
-      unless subsidiaries.count > 0
-        errors.add(:destination_firm_fca_number, :does_not_exist, fca_number: destination_firm_fca_number)
-      end
+      errors.add(:destination_firm_fca_number,
+                 :does_not_exist,
+                 fca_number: destination_firm_fca_number) unless subsidiaries.count > 0
```
Comment (neoeno) [inline app/forms/admin/move_advisers_form.rb]: My original thought, when I commented on this before but didn't actually specify, was:

```
return if subsidiaries.count > 0
errors.add(:destination_firm_fca_number,
                 :does_not_exist,
                 fca_number: destination_firm_fca_number)
```

which is what guard clause means to me (though i've never actually confirmed this is what they actually are)

File: app/helpers/admin/firms_helper.rb [hunk 4]
```diff
@@ -8,13 +8,20 @@ def render_questionnaire_response(firm, attribute_name)
...
   end
+
+  private
+
+  def render_literal_or_fee_or_percentage(value, attribute_name)
+    output = value.to_s
+    output = "£#{output}" if attribute_name =~ /fee/
+    output = "#{output}%" if attribute_name =~ /percent/
+    output
+  end
```
Comment (neoeno) [inline app/helpers/admin/firms_helper.rb]: Much better

File: app/mailers/admin_contact.rb [hunk 5]
```diff
@@ -5,7 +5,8 @@ class AdminContact < ActionMailer::Base
   )
 
   def contact(email, message)
-    @email, @message = email, message
+    @email = email
+    @message = message
```
Comment (neoeno) [inline app/mailers/admin_contact.rb]: Ha — yes.

File: fca_import_scripts/lib/ext_to_sql.rb [hunk 6]
```diff
@@ -127,12 +125,10 @@ def log(str, newline: true)
 
   def write_progress
     @i ||= 0
-    if (@i += 1) % 10_000 == 0
-      log '.', newline: false
-    end
+    log '.', newline: false if (@i += 1) % 10_000 == 0
```
Comment (neoeno) [inline fca_import_scripts/lib/ext_to_sql.rb]: I'd say this is a good candidate for a guard clause as I described before — looks a bit noisy this way.

File: fca_import_scripts/lib/ext_to_sql.rb [hunk 7]
```diff
@@ -127,12 +125,10 @@ def log(str, newline: true)
...
   end
 
   def escape(str)
-    str.gsub("\t", "\\t")
+    str.gsub("\t", '\t')
```
Comment (neoeno) [inline fca_import_scripts/lib/ext_to_sql.rb]: >;(

File: spec/support/error_page.rb [hunk 8]
```diff
@@ -1,4 +1,4 @@
 class ErrorPage < SitePrism::Page
   set_url '/error'
-  set_url_matcher /\/error/
+  set_url_matcher(/\/error/)
```
Comment (neoeno) [inline spec/support/error_page.rb]: I've noticed in places that people have used `%r{/error/}` which seems like a nicer syntax for this case with paths. No need to change it, just something I saw & liked.

**Comment by neoeno** [comment]
I was initially dreading merge conflicts with other WIP branches but I tried one and it looked alright really. Luckily I invalidated my suspicion before complaining! 

:+1:

**Pull request merged**